tests/bsim/bluetooth/mesh/src/test_brg.c

```diff
@@ -0,0 +1,978 @@
+			return;
+		}
+	}
+
+	ASSERT_OK(bt_mesh_brg_cfg_cli_subnet_bridge_set(0, BRIDGE_ADDR,
+							BT_MESH_SUBNET_BRIDGE_ENABLED, &status));
+	if (status != BT_MESH_SUBNET_BRIDGE_ENABLED) {
+		FAIL("Subnet bridge set failed (status %u)", status);
+		return;
+	}
+
+	/* Disable Relay feature to avoid interference in the test. */
+	uint8_t transmit;
+
+	ASSERT_OK(bt_mesh_cfg_cli_relay_set(0, BRIDGE_ADDR, BT_MESH_RELAY_DISABLED,
+					    BT_MESH_TRANSMIT(2, 20), &status, &transmit));
+	if (status) {
+		FAIL("Relay set failed (status %u)", status);
+		return;
+	}
+
+	LOG_INF("Bridge configured");
+}
+
+static void tester_device_configure(uint16_t net_key_idx, uint16_t addr)
+{
+	int err;
+	uint8_t status;
+
+	err = bt_mesh_cfg_cli_app_key_add(net_key_idx, addr, net_key_idx, 0, test_app_key, &status);
+	if (err || status) {
+		FAIL("AppKey add failed (err %d, status %u)", err, status);
+		return;
+	}
+
+	err = bt_mesh_cfg_cli_mod_app_bind(net_key_idx, addr, addr, 0, TEST_MOD_ID, &status);
+	if (err || status) {
+		FAIL("Mod app bind failed (err %d, status %u)", err, status);
+		return;
+	}
+
+	/* Disable SNB on devices to let Subnet Bridge propagate new IV index value. */
+	err = bt_mesh_cfg_cli_beacon_set(net_key_idx, addr, BT_MESH_BEACON_DISABLED, &status);
+	if (err || status) {
+		FAIL("Beacon set failed (err %d, status %u)", err, status);
+		return;
+	}
+
+	LOG_INF("Device 0x%04x configured", addr);
+}
+
+static void tester_ra_cb(uint8_t *data, size_t length)
+{
+	uint8_t type = data[0];
+
+	LOG_HEXDUMP_DBG(data, length, "tester received message");
+
+	ASSERT_TRUE_MSG(length > 1, "Too short message");
+	ASSERT_EQUAL(type, MSG_TYPE_STATUS);
+
+	recvd_msgs_cnt = data[1];
+	ASSERT_EQUAL(recvd_msgs_cnt * sizeof(recvd_msgs[0]), length - 2);
+	memcpy(recvd_msgs, &data[2], recvd_msgs_cnt * sizeof(recvd_msgs[0]));
+
+	k_sem_give(&status_msg_recvd_sem);
+}
+
+static int send_data(uint16_t dst, uint8_t payload)
+{
+	uint8_t data[2] = {MSG_TYPE_DATA, payload};
+
+	return bt_mesh_test_send_ra(dst, data, sizeof(data), NULL, NULL);
+}
+
+static int send_get(uint16_t dst)
+{
+	uint8_t data[1] = {MSG_TYPE_GET};
+
+	return bt_mesh_test_send_ra(dst, data, sizeof(data), NULL, NULL);
+}
+
+struct bridged_addresses_entry {
+	uint16_t addr1;
+	uint16_t addr2;
+	uint8_t dir;
+};
+
+static void bridge_table_verify(uint16_t net_idx1, uint16_t net_idx2, uint16_t start_idx,
+				struct bridged_addresses_entry *list, size_t list_len)
+{
+	struct bt_mesh_bridging_table_list rsp = {
+		.list = NET_BUF_SIMPLE(BT_MESH_RX_SDU_MAX),
+	};
+
+	net_buf_simple_init(rsp.list, 0);
+
+	ASSERT_OK(bt_mesh_brg_cfg_cli_bridging_table_get(0, BRIDGE_ADDR, net_idx1,
+							 net_idx2, start_idx,
+							 &rsp));
+	ASSERT_EQUAL(rsp.status, 0);
+	ASSERT_EQUAL(rsp.net_idx1, net_idx1);
+	ASSERT_EQUAL(rsp.net_idx2, net_idx2);
+	ASSERT_EQUAL(rsp.start_idx, start_idx);
+
+	LOG_HEXDUMP_DBG(rsp.list->data, rsp.list->len, "Received table");
+
+	ASSERT_EQUAL(rsp.list->len / 5, list_len);
```

Comment:
Output of `sizeof` will be 6 because of alignment.

Reply:
I would really prefer to keep the number here because below the entries are pulled from the buffer and I can see that 5 bytes are pulled in total. If it was a constant here, then I would have to go up to the constant definition and check what it defines.